pkg/sql/pg_catalog.go

```diff
@@ -2925,12 +2925,39 @@ var pgCatalogRulesTable = virtualSchemaTable{
 }
 
 var pgCatalogShadowTable = virtualSchemaTable{
-	comment: "pg_shadow was created for compatibility and is currently unimplemented",
-	schema:  vtable.PgCatalogShadow,
+	comment: `pg_shadow lists properties for roles that are marked as rolcanlogin in pg_authid
+https://www.postgresql.org/docs/13/view-pg-shadow.html`,
+	schema: vtable.PgCatalogShadow,
 	populate: func(ctx context.Context, p *planner, _ catalog.DatabaseDescriptor, addRow func(...tree.Datum) error) error {
-		return nil
+		h := makeOidHasher()
+		return forEachRole(ctx, p, func(username security.SQLUsername, isRole bool, noLogin bool, rolValidUntil *time.Time) error {
+			if noLogin {
+				return nil
+			}
+
+			isSuper := tree.DBool(username.IsRootUser() || username.IsAdminRole())
```

Comment:
i just realised on second reading this is wrong (i *think* have to check whether the user is a member of admin), but this is wrong for the pg_user table as well.

maybe worth an issue, but i'm fine with this as is for now (cc @rafiss)

Reply:
oh, nvm, this is right.